mlir/lib/Dialect/LLVMIR/Transforms/InlinerInterfaceImpl.cpp

```diff
@@ -726,8 +726,9 @@ struct LLVMInlinerInterface : public DialectInlinerInterface {
   }
 
   bool isLegalToInline(Operation *op, Region *, bool, IRMapping &) const final {
-    // The inliner cannot handle variadic function arguments.
-    return !isa<LLVM::VaStartOp>(op);
+    // The inliner cannot handle variadic function arguments nor blocktag
+    // operations.
```

Comment:
```suggestion
    // The inliner cannot handle variadic function arguments and blocktag operations prevent inlining since they
    // the blockaddress operations reference them via the callee symbol.
```
nit: tried to clarify a bit what the semantics of the blocktag operation is.